src/software/ai/hl/stp/play/free_kick/free_kick_play_fsm.cpp

```diff
@@ -0,0 +1,213 @@
+#include "free_kick_play_fsm.h"
+
+FreeKickPlayFSM::FreeKickPlayFSM(TbotsProto::AiConfig ai_config)
+    : ai_config(ai_config),
+      best_pass_and_score_so_far(
+          PassWithRating{.pass = Pass(Point(), Point(), 0), .rating = 0}),
+      align_to_ball_tactic(std::make_shared<MoveTactic>()),
+      shoot_tactic(std::make_shared<KickTactic>()),
+      chip_tactic(std::make_shared<ChipTactic>()),
+      passer_tactic(std::make_shared<KickTactic>()),
+      receiver_tactic(std::make_shared<ReceiverTactic>()),
+      offensive_positioning_tactics(std::vector<std::shared_ptr<MoveTactic>>(2)),
+      ranked_zones(std::vector<EighteenZoneId>(18)),
+      pass_generator(
+          PassGenerator<EighteenZoneId>(std::make_shared<const EighteenZonePitchDivision>(
+                                            Field::createSSLDivisionBField()),
+                                        ai_config.passing_config())),
+      MAX_TIME_TO_COMMIT_TO_PASS(Duration::fromSeconds(
+          5)),  // Spend at most 5 seconds looking for a pass. The rule for division B
+                // teams is at most 10 seconds spent on the entire free kick
```

Comment:
we might want to make this a configurable param in `parameters.proto`

Reply:
I think this parameter is quite specific to free kick, so it might be fine to just declare it in FreeKickPlayFSM.h as a static const.